lib/spack/docs/repositories.rst

```diff
@@ -0,0 +1,275 @@
+be more specific when designating, e.g., which package to uninstall:
+
+.. code-block:: console
+
+  spack uninstall llnl.comp.mpich
+
+Or, if you have your own repository registered and you want to **force**
+spack to use the ``builtin`` implementation of ``mpich`` for a certain
+build of ``hdf5``, you could do this:
+
+.. code-block:: console
+
+  spack install hdf5 +mpi ^llnl.comp.mpich
+
+---------------------
+Listing repositories
+---------------------
+
+Spack's :ref:`configuration system <configuration>` allows repository
+settings to come from ``repos.yaml`` files in many locations.  If you
+want to see the repositories registered as a result of all configuration
+files, use ``spack repo list``.
+
+.. _cmd-spack-repo-list:
+
+^^^^^^^^^^^^^^^^^^^
+``spack repo list``
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+  $ spack repo list
+  ==> 2 package repositories.
+  myrepo     /Users/gamblin2/myrepo
+  builtin    /Users/gamblin2/src/spack/var/spack/repos/builtin
+
+Each repository is listed with its associated namespace.
+
+---------------------
+Creating a repository
+---------------------
+
+To make your own repository, you don't need to construct a directory
+yourself; you can use the ``spack repo create`` command.
+
+.. _cmd-spack-repo-create:
+
+^^^^^^^^^^^^^^^^^^^^^
+``spack repo create``
+^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+  $ spack repo create myrepo
+  ==> Created repo with namespace 'myrepo'.
+  ==> To register it with spack, run this command:
+    spack repo add /Users/gamblin2/myrepo
+
+  $ ls myrepo
+  packages/  repo.yaml
+
+  $ cat myrepo/repo.yaml
+  repo:
+    namespace: 'myrepo'
+
+By default, the namespace of a new repo matches its directory's name.
+You can supply a custom namespace with a second argument, e.g.:
+
+  $ spack repo create myrepo llnl.comp
+  ==> Created repo with namespace 'llnl.comp'.
+  ==> To register it with spack, run this command:
+    spack repo add /Users/gamblin2/myrepo
+
+  $ cat myrepo/repo.yaml
+  repo:
+    namespace: 'llnl.comp'
+
+----------------------------------------
+Adding and removing package repositories
+----------------------------------------
+
+.. _cmd-spack-repo-add:
+
+^^^^^^^^^^^^^^^^^^
+``spack repo add``
+^^^^^^^^^^^^^^^^^^
+
+Once your repository is created, you can register it with Spack with
+``spack repo add``:
+
+.. code-block:: console
+
+   $ spack repo add ./myrepo
+   ==> Added repo with namespace 'llnl.comp'.
+
+   $ spack repo list
+   ==> 2 package repositories.
+   llnl.comp    /Users/gamblin2/myrepo
+   builtin      /Users/gamblin2/src/spack/var/spack/repos/builtin
+
+This simply adds the repo to your ``repos.yaml`` file.
+
+Once a repository is registered like this, you should be able to see its
+packages' names in the output of ``spack list``, and you should be able
+to build them using ``spack install <name>`` as you would with any
+builtin package.
+
+.. _cmd-spack-repo-rm:
+
+^^^^^^^^^^^^^^^^^
+``spack repo rm``
```

Comment:
Yeah I can do that.  I even thought about it.  Then I thought about what I would actually type and went with that, but `remove` is clearer.